use pre-created secret instead of embed in syncset

bmeng opened the pull request:
As requested in https://jira.coreos.com/browse/SREP-1831
And discussed with @wanghaoran1988 , he suggested that we should do the clean up things manually on the running clusters.

bmeng commented:
/retest

bmeng commented:
@jewzaam 
I have added some test for the new changes, and also addressed the deletion part.
Please help review again.  Thanks.

jewzaam commented on pkg/controller/deadmanssnitch/deadmanssnitch_controller.go:
```diff
@@ -155,15 +155,47 @@ func (r *ReconcileDeadMansSnitch) Reconcile(request reconcile.Request) (reconcil
+		reqLogger.Info("Secret not found, Creating a new Secret", "Namespace", request.Namespace, "Name", request.Name)
+
+		// snitchName := instance.Spec.ClusterName + "." + instance.Spec.BaseDomain
+		snitchName := "ds-test"
```
```suggestion
		snitchName := instance.Spec.ClusterName + "." + instance.Spec.BaseDomain
```

bmeng replied:
forgot to change it back which is used for local testing...

jewzaam commented on pkg/controller/deadmanssnitch/deadmanssnitch_controller.go:
```diff
@@ -244,21 +275,17 @@ func newSyncSet(namespace string, clusterDeploymentName string, snitchURL string
-								config.KeySnitchURL: []byte(snitchURL),
-							},
+						Source: corev1.ObjectReference{
+							Name:      clusterDeploymentName + config.SecretPostfix,
```
Recommend passing in the source secret name as a function arg rather than reconstructing.

bmeng replied:
moved to function arg

jewzaam commented on pkg/controller/deadmanssnitch/deadmanssnitch_controller.go:
```diff
@@ -292,6 +336,15 @@ func deleteDMS(r *ReconcileDeadMansSnitch, request reconcile.Request, instance *
 
+	// Delete the secret
+	reqLogger.Info("Deleting DMS secret", "Namespace", request.Namespace, "Name", request.Name)
+	err = utils.DeleteSecret(request.Name+config.SecretPostfix, request.Namespace, r.client, reqLogger)
```
Should delete the `Secret` after `SyncSet` simply based on the SS referencing the secret.

bmeng replied:
Changed the order

jewzaam commented on pkg/controller/deadmanssnitch/deadmanssnitch_controller_test.go:
```diff
@@ -460,6 +474,26 @@ func verifyNoSyncSet(c client.Client, expected *SyncSetEntry) bool {
 	return true
 }
 
+func verifyNoSecret(c client.Client, expected *SecretEntry) bool {
```
Take a look again at testing `Secret` deletion.  I commented out this block that deleted the secret and all tests still passed.  I would have expected it to fail.  Not sure why at a glance.

https://github.com/openshift/deadmanssnitch-operator/pull/44/files#diff-cbcc4d0350863bcdb11e774f6cacbaf5R339-R345

bmeng commented:
The reason about why the `delete secret`. I assumed that the test "Test Deleting" should cover that, but actually it does not.

So I updated the test "Test Managed Cluster that later sets noalerts label", and now the deletion of secret should be covered.

And I also renamed some of the function name and arg name for the created secret to "refSecret", to avoid confusion.

jewzaam commented:
/approve
/lgtm
